src/threadnetwork.cs

```diff
@@ -0,0 +1,74 @@
+using CoreFoundation;
+using ObjCRuntime;
+using Foundation;
+
+using System;
+
+namespace ThreadNetwork {
+
+	[iOS (15,0)]
+	[BaseType (typeof (NSObject))]
+	interface THClient
+	{
+		[Async]
+		[Export ("retrieveAllCredentials:")]
+		void RetrieveAllCredentials (Action<NSSet<THCredentials>, NSError> completion);
+
+		[Async]
+		[Export ("deleteCredentialsForBorderAgent:completion:")]
+		void DeleteCredentialsForBorderAgent (NSData borderAgentId, Action<NSError> completion);
+
+		[Async]
+		[Export ("retrieveCredentialsForBorderAgent:completion:")]
+		void RetrieveCredentialsForBorderAgent (NSData borderAgentId, Action<THCredentials, NSError> completion);
+
+		[Async]
+		[Export ("storeCredentialsForBorderAgent:activeOperationalDataSet:completion:")]
+		void StoreCredentialsForBorderAgent (NSData borderAgentId, NSData activeOperationalDataSet, Action<NSError> completion);
+
+		[Async]
+		[Export ("retrievePreferredCredentials:")]
+		void RetrievePreferredCredentials (Action<THCredentials, NSError> completion);
+
+		[Async]
+		[Export ("retrieveCredentialsForExtendedPANID:completion:")]
+		void RetrieveCredentialsForExtendedPanid (NSData extendedPanid, Action<THCredentials, NSError> completion);
```

Comment:
remove `ForExtendedPanid` since it's the first parameter, not the action

Reply:
looking at context / doc [1], the first argument should be `extendedPanId` (uppercase 'I')

[1] _Requests Thread credentials for an extended Personal Area Network (PAN) ID._

Reply:
I missed that.